lib/handlers/auth.post.js

```diff
@@ -1,109 +1,111 @@
-import Metrics from '@metrics/client';
-import abslog from 'abslog';
+import originalUrl from "original-url";
+import HttpError from "http-errors";
+import Metrics from "@metrics/client";
+import abslog from "abslog";
 
-import MultipartParser from '../multipart/parser.js';
-import HttpIncoming from '../classes/http-incoming.js';
-import HttpOutgoing from '../classes/http-outgoing.js';
-import Author from '../classes/author.js';
-import config from '../utils/defaults.js';
+import MultipartParser from "../multipart/parser.js";
+import HttpIncoming from "../classes/http-incoming.js";
+import HttpOutgoing from "../classes/http-outgoing.js";
+import Author from "../classes/author.js";
+import config from "../utils/defaults.js";
+
+/**
+ * @typedef {object} AuthPostOptions
+ * @property {string} [authKey]
+ * @property {string} [cacheControl]
+ * @property {Array<[string, string]>} [organizations] List of key-value pairs [hostname, organization]
+ * @property {import("abslog").AbstractLoggerOptions} [logger]
+ */
 
 const AuthPost = class AuthPost {
-    constructor({
-        organizations,
-        cacheControl,
-        authKey,
-        logger,
-    } = {}) {
-        this._organizations = organizations || config.organizations;
-        this._cacheControl = cacheControl;
-        this._authKey = authKey || config.authKey;
-        this._log = abslog(logger);
-        this._metrics = new Metrics();
-        this._histogram = this._metrics.histogram({
-            name: 'eik_core_auth_post_handler',
-            description:
-                'Histogram measuring time taken in @eik/core AuthPost handler method',
-            labels: {
-                success: true,
-                type: 'unknown',
-            },
-            buckets: [
-                0.005,
-                0.01,
-                0.06,
-                0.1,
-                0.6,
-                1.0,
-                2.0,
-                4.0,
-            ],
-        });
-        this._orgRegistry = new Map(this._organizations);
+	/**
+	 * @param {AuthPostOptions} options
+	 */
+	constructor({ organizations, cacheControl, authKey, logger } = {}) {
+		this._organizations = organizations || config.organizations;
+		this._cacheControl = cacheControl;
+		this._authKey = authKey || config.authKey;
+		this._log = abslog(logger);
+		this._metrics = new Metrics();
+		this._histogram = this._metrics.histogram({
+			name: "eik_core_auth_post_handler",
+			description:
+				"Histogram measuring time taken in @eik/core AuthPost handler method",
+			labels: {
+				success: true,
+				type: "unknown",
+			},
+			buckets: [0.005, 0.01, 0.06, 0.1, 0.6, 1.0, 2.0, 4.0],
+		});
+		this._orgRegistry = new Map(this._organizations);
 
-        this._multipart = new MultipartParser({
-            pkgMaxFileSize: this._pkgMaxFileSize,
-            legalFields: ['key'],
-            sink: this._sink,
-        });
-    }
+		this._multipart = new MultipartParser({
+			legalFields: ["key"],
```

Comment:
Here I removed some fields that would be set as `undefined` since `this._sink` doesn't exist. Might have something to do with it?